docs/keto/modeling/learn-the-concepts.mdx

```diff
@@ -0,0 +1,295 @@
+---
+id: learn-the-concepts
+title: Create your first permission model with OPL
+sidebar_label: Create your first permission model
+---
+
+```mdx-code-block
+import Mermaid from "@site/src/theme/Mermaid"
+```
+
+This section guides you through creating your first permission model using the Ory Permission Language (OPL).
+
+## What is a permission model
+
+Permission checks are answered based on data in the database, for example Bob is owner of document X, as well as the permission
+model, for example all owners of a document can view it.
+
+When you ask Ory Permissions if Bob can view document X, it will check if Bob is owner of the document X. The permission model
+tells Ory Permissions what to check in the database.
+
+## Define a permission model
+
+Designing a permission model is a complex task. Ory Permissions and the Ory Permission Language, as a subset of TypeScript,
+provide a streamlined approach to permissions with the benefit of being processed by a fast, global permission engine.
+
+Just as there is no single approach for programming, there is no universally applicable guide for constructing a permission model.
+Nevertheless, the following iterative process can be a good starting point:
+
+1. Create a list of objects that you want to give access to.
+1. For each object
+   1. Create a list of relationships it has with other objects. In a database, these would be associations expressed with foreign
+      keys.
+   1. Create a list of permissions that you want to check.
+1. Define each relationship in the Ory Permission Language (OPL).
+1. Define each permission in the OPL.
+1. Test your permission model.
+
+## Example: document store
+
+To guide you through the process of defining a permission model, this example will be used:
+
+> We want to model user access to documents such that users can be an owner, editor, or viewer of a document. An owner is also an
+> editor of a document, and an editor is also a viewer of a document. Furthermore, documents can be put into a hierarchy of
+> folders, such that being able to view the parent folder implies being able to view the enclosed folders and documents.
+
+### Create a list of objects
+
+We start with the description and highlight all objects that we find:
+
+> We want to model `user` access to `documents` such that `users` can be an owner, editor, or viewer of a document. An owner is
+> also an editor of a `document`, and an editor is also a viewer of a `document`. Furthermore, `documents` can put into a
+> hierarchy of `folders`, where the ability to view the parent `folder` implies the ability to view the enclosed `folders` and
+> `documents`.
+
+We found these objects:
+
+- User
+- Document
+- Folder
```

Comment:
Somehow it looks weird when we talk about users as objects.

```suggestion
### Create a list of objects and subjects

We start with the description and highlight all **objects** and _subjects_ that we find:

> We want to model _user_ access to **documents** such that _users_ can be an owner, editor, or viewer of a document. An owner is
> also an editor of a **document**, and an editor is also a viewer of a **document**. Furthermore, **documents** can put into a
> hierarchy of **directories**, where the ability to view the parent **directory** implies the ability to view the enclosed **directories** and
> **documents**.

Therefore, we have the `User` subject and `Document` and `Directory` objects.
```

see [folders vs directories](https://www.baeldung.com/cs/directories-vs-folders).